contrib/mesos/pkg/scheduler/components/podreconciler/podreconciler.go

```diff
@@ -81,11 +81,9 @@ func (s *podReconciler) Reconcile(t *podtask.T) {
 	}
 
 	log.Infof("pod %v scheduled on %q according to apiserver", pod.Name, pod.Spec.NodeName)
-	if t.Spec.AssignedSlave != pod.Spec.NodeName {
+	if t.Has(podtask.Launched) && !t.Has(podtask.Bound) {
```

Comment:
Also note here that the if-then-else case are shuffled due to the new logic. 

In fact the code which is left triggers a relaunch when it fails on one node. I am not even sure this logic is worth it. Which actual use-cases do we want to cover here?